File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/ConEditoras.pas
unit ConEditoras;

interface

uses
  System.Drawing, wx, Consultas;

type
  TfrmConEditoras = class(TfrmConsultas)
  private
    procedure OnClose(sender: System.Object; e: wx.Event);
    procedure OnBtnFechar(sender: System.Object; e: wx.Event);
    procedure OnBtnLimpar(sender: System.Object; e: wx.Event);
    procedure OnBtnPesquisar(sender: System.Object; e: wx.Event);
    procedure OnGridSelectCell(sender: System.Object; e: wx.Event);
    procedure OnGridLabelLeftClick(sender: System.Object; e: wx.Event);
  public
    constructor Create(parent: wx.Window);
  end;

var
  frmConEditoras: TfrmConEditoras;

  [assembly: RuntimeRequiredAttribute(TypeOf(TfrmConEditoras))]

implementation

uses Resource, ConsultasSQL;

constructor TfrmConEditoras.Create(parent: wx.Window);
begin
  inherited Create(parent, 'Editora:');

  Title := Res.STR_CONEDITORA;
  //labDescricao.Label := 'Editora:';
  gridConsulta.SetColLabelValue(1, 'Editora');

  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, OnBtnPesquisar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, OnBtnLimpar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, OnBtnFechar );
  EVT_CLOSE( OnClose );
  EVT_GRID_SELECT_CELL( OnGridSelectCell );
  EVT_GRID_LABEL_LEFT_CLICK( OnGridLabelLeftClick );
end;

procedure TfrmConEditoras.OnClose(sender: System.Object; e: wx.Event);
begin
  Fecha(Res.MNU_CON_EDITORAS);
	e.Skip();
end;

procedure TfrmConEditoras.OnBtnFechar(sender: System.Object; e: wx.Event);
begin
  Fecha(0);
  e.Skip();
end;

procedure TfrmConEditoras.OnBtnLimpar(sender: System.Object; e: wx.Event);
begin
  LimparDados();
  e.Skip();
end;

procedure TfrmConEditoras.OnBtnPesquisar(sender: System.Object; e: wx.Event);
begin
  PesquisarDados(ConsSQL.Editora('S','', edtDescricao.Value),
    Res.STR_EDTNENC, 2);
  e.Skip();
end;

procedure TfrmConEditoras.OnGridSelectCell(sender: System.Object; e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

procedure TfrmConEditoras.OnGridLabelLeftClick(sender: System.Object;
  e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

end.

File: LivrusVirtus/wxWindows/wx.NET/VCPP.NET/ConEditoras.cpp
#include "stdafx.h"
#include "ConEditoras.h"
#include "Resource.h"
#include "ConsultasSQL.h"

namespace Livrus
{
  frmConEditoras::frmConEditoras( wx::Window* parent )
  	: frmConsultas(parent) {

    Title = Resource::STR_CONEDITORA;
    labDescricao->Label = "Editora:  ";
    gridConsulta->SetColLabelValue(1, "Editora");

    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNPESQUISAR, 
      new wx::EventListener( this, OnBtnPesquisar ));
    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNLIMPAR, 
      new wx::EventListener( this, OnBtnLimpar ));
    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNFECHAR, 
      new wx::EventListener( this, OnBtnFechar ));
  	EVT_CLOSE( new wx::EventListener( this, OnClose ) );
    EVT_GRID_SELECT_CELL(new wx::EventListener( this, OnGridSelectCell ));
    EVT_GRID_LABEL_LEFT_CLICK(new wx::EventListener( this, 
      OnGridLabelLeftClick ));
  }

  Void frmConEditoras::OnClose( System::Object* sender, wx::Event* e ){
    Fecha(Resource::MNU_CON_EDITORAS);
  	e->Skip();
  }

  Void frmConEditoras::OnBtnFechar( System::Object* sender, wx::Event* e ){
    Fecha(0);
    e->Skip();
  }
  
  Void frmConEditoras::OnBtnLimpar( System::Object* sender, wx::Event* e ){
    LimparDados();
    e->Skip();
  }
  
  Void frmConEditoras::OnBtnPesquisar( System::Object* sender, wx::Event* e ){
    PesquisarDados(ConsultasSQL::ConsSQL->Editora('S',"", edtDescricao->Value),
      Resource::STR_EDTNENC, 2);
    e->Skip();
  }
  
  Void frmConEditoras::OnGridSelectCell( System::Object* sender, 
    wx::Event* e ){
    wx::GridEvent* evt = dynamic_cast<wx::GridEvent*>(e);
    setLabRegistros(evt->Row);
    e->Skip();
  }
  
  Void frmConEditoras::OnGridLabelLeftClick( System::Object* sender, 
    wx::Event* e ){
    wx::GridEvent* evt = dynamic_cast<wx::GridEvent*>(e);
    setLabRegistros(evt->Row);
    e->Skip();
  }
}
